fgpyo/fasta/reference_set_builder.py

```diff
@@ -0,0 +1,174 @@
+----------------------------------------------------------------
+"""
+# import hashlib
+import os
+import textwrap
+from tempfile import NamedTemporaryFile
+from typing import List
+from typing import Optional
+
+# pylint: disable=W0511
+
+
+class ReferenceSetBuilder:
+    """
+    Builder for constructing one or more fasta records.
+    """
+
+    # The default asssembly
+    DEFAULT_ASSEMBLY: str = "testassembly"
+
+    # The default species
+    DEFAULT_SPECIES: str = "testspecies"
+
+    # Way to store instance of ReferenceBuilder
+    # TODO make something better than a list... probably
+    REF_BUILDERS: List["ReferenceBuilder"] = []
+
+    def __init__(
+        self,
+        assembly: Optional[str] = None,
+        species: Optional[str] = None,
+        line_length: int = 80,
+    ):
+        self.assembly: str = assembly if assembly is not None else self.DEFAULT_ASSEMBLY
+        self.species: str = species if species is not None else self.DEFAULT_SPECIES
+        self.line_length = line_length
+
+    def add(
+        self,
+        name: str,
+    ) -> "ReferenceBuilder":
+        """
+        Returns instance of ReferenceBuilder
+        """
+
+        builder = ReferenceBuilder(name=name, assembly=self.assembly, species=self.species)
+        self.REF_BUILDERS.append(builder)
+        return self.REF_BUILDERS[-1]
+
+    def writer_helper(
+        self,
+    ) -> None:
+        """ Place holder for helper funciton """
+        return None
+
+    def to_temp_file(
+        self,
+        delete_on_exit: Optional[bool] = True,
+        calculate_md5_sum: Optional[bool] = False,
+    ) -> None:
+        """
+        For each instance of ReferenceBuilder in REF_BUILDERS write record to temp file
+        """
+        # Write temp file path
+        path = NamedTemporaryFile(
+            prefix=f"{self.assembly}_{self.species}",
+            suffix=".fasta",
+            delete=delete_on_exit,
+            mode=("a+t"),
+        )
+
+        for record in range(len(self.REF_BUILDERS)):
+            seq = self.REF_BUILDERS[record].sequences
+            assembly = self.REF_BUILDERS[record].assembly
+            species = self.REF_BUILDERS[record].species
+            name = self.REF_BUILDERS[record].name
+            header = f">{name}[{assembly}][{species}]\n"
+            seq_format = "\n".join(textwrap.wrap(seq, self.line_length))
+            try:
+                path.write(header)
+                path.write(f"{seq_format}\n\n")
+            except OSError as error:
+                print(f"{error}\nCound not write to {path}")
+
+        # if calculate_md5_sum:
+        # pylint: disable=W0612
+        # with open(path.name, "rb") as path_to_read:
+        # contents = path_to_read.read()
+        # md5 = hashlib.md5(contents).hexdigest()
+
+        # Use md5 to write dict
+        # Write .fai
+
+        path.close()
+
+    def to_file(
+        self,
+        path: str,
```

Comment:
use `Path` from `pathlib`
```suggestion
        path: Path,
```

Reply:
Pathlib is used in commit cb8ac832a05ba4a3ad304edb2de19c57d6f9b8f3